src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -348,6 +350,33 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
         auto pass_config = manager.get_pass_config();
         manager.set_per_pass_validation(false);
 
+        // Transformation of SDPA to VLSDPA for QWen2.x-VL,
+        // Note: this should be applied before TransposeFusion.
+        manager.register_pass<ov::pass::SDPAToVLSDPA>();
+        // Disable SDPAToVLSDPA if XMX architectures is unavaiable or IGC incompatiable.
+        pass_config->set_callback<ov::pass::SDPAToVLSDPA>(
+                [&](const_node_ptr &) -> bool {
+                    auto& engine = m_context->get_engine();
+                    const auto& info = engine.get_device_info();
+                    if (!(info.arch >= cldnn::gpu_arch::xe_lp)) {
+                        return true;
+                    }
+
+                    if (!config.get_use_cm()) {
+                        util::log_message("You may miss SDPAToVLSDPA optimization for QWenVL model,"
```

Comment:
nit:
maybe we can remove this check? It's enabled by default and not directly exposed to users

Reply:
It is for debug purpose I think.